src/components/EventUserTable.js

```diff
@@ -27,25 +40,111 @@ export class EventUserTable extends Component {
     this.getEventTableData(this.props.event.id);
   }
 
-  async getEventTableData(eventID) {
+  /* updates stats and the rows in the table
+     faculty, gender, dietary, and year stats are only computed on the initial render of the component
+     # of registered/checkedin etc. is computed every single time this function is called
+  */
+  async getEventTableData(eventID, firstRender) {
+
     const params = new URLSearchParams({
+      eventID: eventID
+    })
+
+    await fetchBackend('/registration/scanEvent?' + params, 'GET')
+      .then(response => response.json())
+      .then(response => {
+        let heardFrom = {};
+        for (let i = 0; i < response.size; ++i) {
+          const temp = response.data[i];
+          if (temp.hasOwnProperty('heardFrom')) {
+            heardFrom[temp.heardFrom] = heardFrom.hasOwnProperty(temp.heardFrom) ? heardFrom[temp.heardFrom] + 1 : 1;
+          }
+        }
+        this.setState({ heardFrom })
+      })
+
+    const eventParams = new URLSearchParams({
       id: eventID
     });
 
-    fetchBackend('/events/getUsers?' + params, 'GET')
+    fetchBackend('/events/getUsers?' + eventParams, 'GET')
       .then(response => response.json())
-      .then(response => {
-        this.setState({ rows: response });
+      .then(async users => {
+        let obj = {
+          'registered': 0,
+          'checkedIn': 0,
+          'waitlisted': 0,
+          'cancelled': 0,
```

Comment:
Maybe we can make the `registrationStatus` an object as well since the other fields are objects